tap20.md

```diff
@@ -0,0 +1,355 @@
+        "keyids": [
+          "1bf1c6e3cdd3d3a8420b19199e27511999850f4b376c4547b2f32fba7e80fca3"
+        ],
+        "threshold": 1
+      }
+    },
+    "version": 1
+  }
+}
+```
+
+**NB: `keyids` used in `root.json` are sha256 hashes of the canonical JSON form of `keyval` to keep them unique within the document and do not correspond to the `kid` field used in OIDC resources**
+
+When OIDC providers' public keys are rotated, the old public keys in `root.json` can no longer be used to verify new signatures. Similarly, the new public keys cannot be used to verify signatures under the old public keys in `root.json`.
+
+The Timestamp role is responsble for maintaining a log of OIDC OP public keys for all issuers of `opk` type keys in `root.json`. To keep the size of the timestamp.json metadata small, only one additional entry is added: `opkl.json`.
+
+The timestamp role in this example is signing using OpenPubkey, but this isn't necessary in order to maintain the log for other roles; offline keys could also have been used.
+
+`timestamp.json`
+
+```json
+{
+  "signatures": [
+    {
+      "keyid": "1bf1c6e3cdd3d3a8420b19199e27511999850f4b376c4547b2f32fba7e80fca3",
+      "sig": "...OPK..."
+    }
+  ],
+  "signed": {
+    "_type": "timestamp",
+    "spec_version": "1.0.0",
+    "expires": "2030-01-01T00:00:00Z",
+    "meta": {
+      "snapshot.json": {
+        "hashes": {
+          "sha256": "c14aeb4ac9f4a8fc0d83d12482b9197452f6adf3eb710e3b1e2b79e8d14cb681"
+        },
+        "length": 1007,
+        "version": 1
+      },
+      "opkl.json": {
+        "hashes": {
+          "sha256": "4980db466497a606351072cf9afe551b1b236b2b0745316c3e3cbcf756040469"
+        },
+        "length": 6008,
+        "version": 1
+      }
+    },
+    "version": 1
+  }
+}
+
+```
+
+The `keyid` above is a reference to the corresponding `keyid` in `root.json`, but because OIDC public keys rotate, the actual public key used in the signature changes over time as the keys are rotated. Resolution of public keys and use of signatures is discussed below.
+
+`opkl.son` contains a map of JWK thumbprints (calculated according to [RFC-7638](https://datatracker.ietf.org/doc/html/rfc7638)) to JWK metadata, including the `jwk` whose contents are as returned from the JWKS endpoint.
+
+Until the first timestamp, the file is empty.
+
+### Updating the OIDC OP key log
+
+`opkl.json` is updated by the following process for each issuer in `root.json`:
+
+* Initial State: root retrieves and stores all OP public keys (A, B) aka, the primordial set
+* T1: timestamp role retrieves all OP public keys (A, B). Compares to those stored in `root.json` and fails if they don't match.  Requests an ID token from issuer, and determines which OP public key it is associated it (say B). Checks that it's in the primoridial set, and fails otherwise. Creates `opkl.json` containing both keys A and B (see below) including a signature (using GQ?) over A. Then `timestamp.json` is updated and signed.
```

Comment:
Would this work for multiple identity providers or would the `timestamp` role's OP have to match the public key log's OP?

It would be interesting to see if/how this could provide OP key logs for multiple providers in the same TUF repo. I suppose you could add multiple key log files to the `timestamp.json` metadata (one per provider).